// khalti-epayment/dist/index.mjs
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// index.ts
import axios from "axios";
function awTry(promise) {
  return __async(this, null, function* () {
    try {
      const data = yield promise;
      return [data, null];
    } catch (err) {
      console.error(err);
      return [null, err];
    }
  });
}
var paymentLookup = (instance) => {
  return (_0) => __async(void 0, [_0], function* ({ pidx }) {
    const [response, error] = yield awTry(
      instance.get(`/epayment/lookup?pidx=${pidx}`)
    );
    if (error) {
      console.log("Error, in lookupPayment: ", error);
    }
    return response.data;
  });
};
var initiatePayment = (instance) => {
  return (payload) => __async(void 0, null, function* () {
    const [response, error] = yield awTry(
      instance.post("/epayment/initiate/", payload)
    );
    if (error) {
      console.log("Error, in initiatePayment: ", error);
    }
    return response.data;
  });
};
var createKhaltiInstance = ({
  env,
  secretKey
}) => {
  const urls = {
    sandbox: "https://a.khalti.com/api/v2",
    production: "https://khalti.com/api/v2"
  };
  const instance = axios.create({
    baseURL: urls[env],
    headers: {
      "Content-Type": "application/json",
      Authorization: `Key ${secretKey}`
      // live secret key
    },
    responseType: "json"
  });
  return {
    paymentLookup: paymentLookup(instance),
    initiatePayment: initiatePayment(instance)
  };
};
export {
  createKhaltiInstance
};
